### PR TITLE
the return of player character spawners (feat. mindshield bullshit)

### PR DESCRIPTION
**Pull request by MilonPL**
## About the PR

ghost role player character spawners are back from the dead (get it???)
idfk how it ended up with HideSpawnMenu, but we use these
removed the do not map suffix to fit with other spawners

## Breaking changes

**Changelog**

no cl i'll just ping everyone in admin general :blunt:

### CODE REVIEW

File: Content.Server/DeltaV/Ghost/Roles/Components/GhostRoleCharacterSpawnerComponent.cs [hunk 1]
```diff
@@ -7,16 +7,28 @@
...
+        ///  Whether to give the MindShield and AntagImmune components on spawn.
+        /// </summary>
+        [DataField]
+        public bool MindShield;
+
+        /// <summary>
+        /// Whether to give the TargetObjectiveImmune component on spawn.
+        /// </summary>
+        [DataField]
+        public bool TargetObjectiveImmune;
```
Comment (deltanedas) [inline Content.Server/DeltaV/Ghost/Roles/Components/GhostRoleCharacterSpawnerComponent.cs]: replace this with a ComponentRegistry

File: Content.Server/DeltaV/Ghost/Roles/Components/GhostRoleCharacterSpawnerComponent.cs [hunk 2]
```diff
@@ -7,16 +7,28 @@
...
-        [ViewVariables(VVAccess.ReadWrite)] [DataField("availableTakeovers")]
+        [DataField]
         public int AvailableTakeovers = 1;
 
         [ViewVariables]
         public int CurrentTakeovers = 0;
 
-        [ViewVariables(VVAccess.ReadWrite)] [DataField("outfitPrototype")]
+        [DataField]
         public string OutfitPrototype = "PassengerGear";
```
Comment (deltanedas) [inline Content.Server/DeltaV/Ghost/Roles/Components/GhostRoleCharacterSpawnerComponent.cs]: ProtoId

File: Content.Server/DeltaV/Ghost/Roles/GhostRoleSystem.Character.cs [hunk 3]
```diff
@@ -45,6 +47,15 @@ private void OnSpawnerTakeCharacter( EntityUid uid, GhostRoleCharacterSpawnerCom
...
 
+            if (component.MindShield)
+            {
+                EnsureComp<MindShieldComponent>(mob);
+                EnsureComp<AntagImmuneComponent>(mob);
+            }
+
+            if (component.TargetObjectiveImmune)
+                EnsureComp<TargetObjectiveImmuneComponent>(mob);
+
```
Comment (deltanedas) [inline Content.Server/DeltaV/Ghost/Roles/GhostRoleSystem.Character.cs]: have this be AddComponents(mob, component.AddComponents)

File: Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml [hunk 4]
```diff
@@ -17,6 +17,25 @@
...
           state: full
 
+- type: entity
+  parent: SpawnPointPlayerCharacter
+  id: SpawnPointPlayerCharacterMindShield
+  name: ghost role spawn point
+  suffix: mindshielded, player character
+  components:
+  - type: GhostRoleCharacterSpawner
+    mindShield: true
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml]:
```
addedComponents:
- MindShield
- AntagImmune
```

File: Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml [hunk 5]
```diff
@@ -17,6 +17,25 @@
...
+
+- type: entity
+  parent: SpawnPointPlayerCharacter
+  id: SpawnPointPlayerCharacterTargetImmune
+  name: ghost role spawn point
+  suffix: mindshielded, KillObjectiveImmune, player character
+  components:
+  - type: GhostRoleCharacterSpawner
+    mindShield: true
+    targetObjectiveImmune: true
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml]:
```
addedComponents:
- MindShield
- AntagImmune
- TargetObjectiveImmune
```

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml (hunk 5 above)
Comment (MilonPL) [reply Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml]: i wanted it to be easily VV'able but yeah real

Comment (deltanedas) [reply Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml]: would be true if there werent prototypes but there are, so no reason not to do this (if you are going to VV something just use a different prototype)
this is more flexible for prototypes and will be infinitely more flexible for VV if/when list gets vv writing support

Comment (MilonPL) [reply Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml]: I mean I have loadprototype and SCSI anyways and the admins can just cope so yeah

**Review by deltanedas — DISMISSED** [review]

File: Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml [hunk 6]
```diff
@@ -17,6 +17,28 @@
...
 
+- type: entity
+  parent: SpawnPointPlayerCharacter
+  id: SpawnPointPlayerCharacterMindShield
+  name: ghost role spawn point
+  suffix: mindshielded, player character
+  components:
+  - type: GhostRoleCharacterSpawner
+    addedComponents:
+    - MindShield
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Markers/Spawners/ghost_roles.yml]: just put `type: ` before all these

**Review by deltanedas — APPROVED** [review]

**Pull request merged**